packages/oc-server-compiler/index.js

```diff
@@ -9,42 +9,55 @@ const webpackConfigurator = require('oc-webpack').configurator;
 const MemoryFS = require('memory-fs');
 
 module.exports = (options, callback) => {
-  const serverFileName = options.componentPackage.oc.files.data;
+  const { componentPackage, production, publishPath } = options;
+
+  const serverFileName = componentPackage.oc.files.data;
   const serverPath = path.join(options.componentPath, serverFileName);
   const publishFileName = options.publishFileName || 'server.js';
-  const publishPath = options.publishPath;
   const stats = options.verbose ? 'verbose' : 'errors-only';
-  const dependencies = options.componentPackage.dependencies || {};
+  const dependencies = componentPackage.dependencies || {};
 
   const config = webpackConfigurator({
-    serverPath,
-    publishFileName,
     dependencies,
+    production,
+    publishFileName,
+    serverPath,
     stats
   });
 
   async.waterfall(
     [
       next => compiler(config, next),
       (data, next) => {
-        const compiledServer = new MemoryFS(data).readFileSync(
-          '/build/server.js',
-          'UTF8'
-        );
-        return fs.ensureDir(publishPath, err => next(err, compiledServer));
+        const basePath = path.join(serverPath, '../build');
```

Comment:
I need MemoryFS to write to a specific path, so that the sourcemap reference to server.js resolves correctly (via the `sourceRoot` setting).